script/licenses.py

```diff
@@ -29,23 +39,60 @@ class PackageDefinition:
     """Package definition."""
 
     license: str
+    license_expression: str | None
+    license_metadata: str | None
+    license_classifier: list[str]
     name: str
     version: AwesomeVersion
 
     @classmethod
     def from_dict(cls, data: PackageMetadata) -> PackageDefinition:
         """Create a package definition from PackageMetadata."""
         if not (license_str := "; ".join(data["license_classifier"])):
-            license_str = (
-                data["license_metadata"] or data["license_expression"] or "UNKNOWN"
-            )
+            license_str = data["license_metadata"] or "UNKNOWN"
         return cls(
             license=license_str,
+            license_expression=data["license_expression"],
+            license_metadata=data["license_metadata"],
+            license_classifier=data["license_classifier"],
             name=data["name"],
             version=AwesomeVersion(data["version"]),
         )
 
 
+# Incomplete list of OSI approved SPDX identifiers
+# Add more as needed, see https://spdx.org/licenses/
+OSI_APPROVED_LICENSES_SPDX = {
+    "0BSD",
+    "AFL-2.1",
+    "AGPL-3.0-only",
+    "AGPL-3.0-or-later",
+    "Apache-2.0",
+    "BSD-1-Clause",
+    "BSD-2-Clause",
+    "BSD-3-Clause",
+    "EPL-1.0",
+    "EPL-2.0",
+    "GPL-2.0-only",
+    "GPL-2.0-or-later",
+    "GPL-3.0-only",
+    "GPL-3.0-or-later",
+    "HPND",
+    "ISC",
+    "LGPL-2.1-only",
+    "LGPL-2.1-or-later",
+    "LGPL-3.0-only",
+    "LGPL-3.0-or-later",
+    "MIT",
+    "MPL-1.1",
+    "MPL-2.0",
+    "PSF-2.0",  # not approved
+    "Python-2.0.1",  # not approved
```

Comment:
Are these two licenses not approved by the OSI? Why do we include them here?

Reply:
On the SPDX website `PSF-2.0` is listed as `not` approved. https://spdx.org/licenses/
However, that might be outdated. According to Wikipedia and the OSI website, it is approved.
https://en.wikipedia.org/wiki/Python_Software_Foundation_License
https://opensource.org/license/python-2-0

Although there is also some discussion that the OSI website is has the wrong SPDX identifier. It's quite a mess TBH.
https://github.com/spdx/license-list-XML/issues/2197

Anyhow, `PSF-2.0` is used as license for Python itself and many packages which copy / reuse parts of the stdlib also use it as license, so I think it's fine / we should keep it.

--
Regarding `Python-2.0.1`, that's the license used by Python 2. It isn't currently used by any packages, so I removed it the list.